jerry-core/ecma/operations/ecma-regexp-object.h

```diff
@@ -29,26 +29,30 @@
  * @{
  */
 
-#define RE_EXECUTE_RECURSION_LIMIT  1000  /* Limit of RegExp executor recursion depth */
-#define RE_EXECUTE_MATCH_LIMIT      10000 /* Limit of RegExp execetur matching steps */
+/**
+ * Limit of RegExp executor recursion depth
+ */
+#define RE_EXECUTE_RECURSION_LIMIT  1000
```

Comment:
It is used in `re_match_regexp` in `ecma-regexp-object.cpp`. It is used to avoid too deep stack calls and bail out with a range error. It also makes sure that an infinite loop never occurs in the executor, so the engine never stuck.